python/extract_docstring.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import re
+import inspect
+import six
+import json
+import xgboost
+import sqlflow_models
+from tensorflow.estimator import *
+
+def parse_ctor_args(f, prefix=''):
+  r"""Given an class or function, parse the class constructor/function details
+  from its docstring
+
+  For example, the docstring of sqlflow_models.DNNClassifier.__init__ is:
+  '''DNNClassifier
+  :param feature_columns: feature columns.
+  :type feature_columns: list[tf.feature_column].
+  :param hidden_units: number of hidden units.
+  :type hidden_units: list[int].
+  :param n_classes: List of hidden units per layer.
+  :type n_classes: int.
+  '''
+  Calling parse_ctor_args(sqlflow_models.DNNClassifier, ":param") returns:
+  {
+      "feature_columns": "feature columns. :type feature_columns: list[tf.feature_column].",
+      "hidden_units": "number of hidden units. :type hidden_units: list[int].",
+      "n_classes": "List of hidden units per layer. :type n_classes: int."
+  }
+  And calling parse_ctor_args(parse_ctor_args) returns:
+  {
+      "f": "The class or function whose docstring to parse",
+      "prefix": "The prefix of parameters in the docstring"
+  }
+
+  Args:
+    f: The class or function whose docstring to parse
+    prefix: The prefix of parameters in the docstring
+
+  Returns:
+    A dict with parameters as keys and splitted docstring as values
+  """
+
+  try:
+      doc = f.__init__.__doc__
```

Comment:
inconsistent indent

Reply:
Thanks for finding this, I'll fix it in the next commit.